bin/check-instance-cpucredits.rb

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env ruby
+#
+# check-instance-cpucredits
+#
+# DESCRIPTION:
+#   This plugin checks the CPU credit balance of the instance it is ran on
+#   and alerts if they are low.
+#
+# OUTPUT:
+#   plain-text
+#
+# PLATFORMS:
+#   Linux
+#
+# DEPENDENCIES:
+#   gem: aws-sdk
+#   gem: sensu-plugin
+#
+# USAGE:
+#
+#   Check the current instance's CPU credits and alert critical if below 10 and warning if below 50 and greater than 10:
+#   check-instance-cpucredits.rb
+#
+#   Check the specified instance's CPU credits and alert critical if below 10 and warn below 50 (must specify region):
+#   check-instance-cpucredits.rb -i i-0d2fe2a5f09338fef -r eu-west-1
+#
+#   Check instance CPU credits and warn if they are less than 100 and alert critical if below 10:
+#   check-instance-cpucredits.rb -w 100 -c 10
+#
+#   Check instance CPU credits:
+#   check-instance-cpucredits.rb -w 100 -c 10
+#
+# LICENSE:
+#    MIT License
+#
+#    Copyright (c) 2017 Claranet
+#
+#    Permission is hereby granted, free of charge, to any person obtaining a copy
+#    of this software and associated documentation files (the "Software"), to deal
+#    in the Software without restriction, including without limitation the rights
+#    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#    copies of the Software, and to permit persons to whom the Software is
+#    furnished to do so, subject to the following conditions:
+#
+#    The above copyright notice and this permission notice shall be included in all
+#    copies or substantial portions of the Software.
+#
+#    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+#    SOFTWARE.
+#
+
+require 'net/http'
+require 'aws-sdk'
+require 'sensu-plugin/check/cli'
+
+class CheckInstanceCpuCredits < Sensu::Plugin::Check::CLI
+  VERSION = '0.0.1'.freeze
```

Comment:
We do not typically version the actual checks, I am not opposed to it but I'd like to hear a good argument for it. ping @eheydrick

Reply:
I copied this from another check, I will remove it.